roc_jni/src/main/cpp/endpoint.cpp

```diff
@@ -0,0 +1,211 @@
+#include "org_rocstreaming_roctoolkit_Endpoint.h"
+#include "common.h"
+#include "endpoint.h"
+
+#include <roc/config.h>
+
+#define ENDPOINT_CLASS              PACKAGE_BASE_NAME "/Endpoint"
+#define PROTOCOL_CLASS              PACKAGE_BASE_NAME "/Protocol"
+
+int endpoint_unmarshal(JNIEnv *env, roc_endpoint** endpoint, jobject jendpoint) {
+    jclass          endpointClass;
+    jclass          protocolClass;
+    jobject         tempObject;
+    jstring         jstr;
+    roc_protocol    protocol;
+    const char*     host;
+    int             port;
+    const char*     resource;
+    char            err = 0;
+
+    if (jendpoint == NULL)
+        return -1;
+
+    endpointClass = env->FindClass(ENDPOINT_CLASS);
+    assert(endpointClass != NULL);
+
+    protocolClass = env->FindClass(PROTOCOL_CLASS);
+    assert(protocolClass != NULL);
+
+    if (roc_endpoint_allocate(endpoint) != 0) return -1;
+
+    tempObject = get_object_field(env, endpointClass, jendpoint, "protocol", "L" PROTOCOL_CLASS ";");
+    protocol = (roc_protocol) get_enum_value(env, protocolClass, tempObject);
+    if (roc_endpoint_set_protocol(*endpoint, protocol) != 0) return -1;
+
+    jstr = (jstring) get_object_field(env, endpointClass, jendpoint, "host", "Ljava/lang/String;");
+    host = env->GetStringUTFChars(jstr, 0);
+    assert(host != NULL);
```

Comment:
Why here we do assertion, but below we do a check:

```c
if (jstr != NULL) {
```

?

Reply:
`host` can't be null and it's checked on java side. But `resource` could be null and it's ok